swagger1/src/main/java/com/cerner/beadledom/swagger/SwaggerUiResource.java

```diff
@@ -66,4 +71,21 @@ public Response getSwaggerUiAsset(@PathParam("asset") String assetPath) {
     InputStream stream = getClass().getResourceAsStream("ui-dist/" + assetPath);
     return Response.ok(stream).build();
   }
+
+  private boolean isRequestSecure(SecurityContext securityContext, HttpHeaders httpHeaders) {
```

Comment:
A thought occurred to me. Is the correct scheme already on the `UriInfo` regardless of being behind a reverse proxy or not? I'm guessing the UriInfo would have the correct domain rather than some local domain that is part of a DCOS cluster, and if it has the correct domain it seems like it would have the correct scheme as well.

Reply:
From the UriInfo docs, `the URIs are relative to the base URI of the application`.  So I think it would contain the correct scheme (that the base application used) as well. 

Would there be any good way to test this besides dropping this in an application and making sure it runs?

https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/UriInfo.html

Reply:
Some Google searches make it seem like it might not be as easy as I was hoping. I think the only way to test will be to actually deploy a service behind a reverse proxy and try it.

Reply:
😢

Reply:
Well I tested it with a service running locally under `https` and `http` and it worked in those two environments. https://github.com/cerner/beadledom/pull/137/commits/418167b1919359a438b0d8d96aab8aa55eb24e9f